internal/rest/resources/resources.go

```diff
@@ -124,3 +133,19 @@ func ValidateEndpoints(extensionServers map[string]rest.Server, coreAddress stri
 
 	return nil
 }
+
+// "core/"   -> "core"
+// "/core"   -> "core"
+// "/core/x" -> "core"
+// "/"       -> ""
+// Returns the first non-root path element in `path`.
+func firstPathElement(path string) string {
+	dir, file := filepath.Split(path)
```

Comment:
I understand his recursively traverses the path from right to left?

I am wondering if it would be faster to "just" split the path on `/` and take either the first or second element of the slice depending on if it's a relative or absolute path?

Reply:
Maybe with a `strings.SplitN( , , 3) to only ever look at the first elements?

Reply:
I wasn't concerned about speed given the application, and was hoping to use a library function that was tied to path semantics instead of generic strings. Went ahead and switched it though.